hack/build_catalog.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```

Comment:
Slightly improved version

1. added additional variables
2. variables and command output are enclosed in double quotes
3. added regex to avoid hardcoded values in CSV

```suggestion
#!/bin/bash

set -euo pipefail

DEFAULT_PLATFORM="linux/amd64"
#export USER=<quay_user>  # Set quay user for personal builds
VERSION="0.0.1"    # Set the version
QUAY_USER="${USER:-openshift}"

IMAGE_TAG_BASE="quay.io/${QUAY_USER}/lightspeed-operator"
BUNDLE_IMAGE="quay.io/${QUAY_USER}/lightspeed-operator-bundle:v${VERSION}"
CATALOG_IMAGE="quay.io/${QUAY_USER}/lightspeed-catalog:${VERSION}"

CSV_FILE="bundle/manifests/lightspeed-operator.clusterserviceversion.yaml"
CATALOG_FILE="lightspeed-catalog/operator.yaml"
CATALOG_DOCKER_FILE="lightspeed-catalog.Dockerfile"

echo "Building File Based Catalog (FBC) for ${USER}"
echo "Set IMAGE_TAG_BASE: ${IMAGE_TAG_BASE}"

rm -rf ./bundle
make bundle VERSION="${VERSION}"

sed -i.bak "s/name: lightspeed-operator.v[0-9].[0-9].[0-9]/name: lightspeed-operator.v${VERSION}/" "${CSV_FILE}"
sed -i.bak "s/version: [0-9].[0-9].[0-9]/version: ${VERSION}/" "${CSV_FILE}"
rm -rf "${CSV_FILE}.bak"

make bundle-build bundle-push VERSION="${VERSION}"
echo "Built bundle image ${BUNDLE_IMAGE}"

echo "Adding bundle image to FBC using image ${BUNDLE_IMAGE}" 
bin/opm render "${BUNDLE_IMAGE}" --output=yaml >> "${CATALOG_FILE}"

cat << EOF >> "${CATALOG_FILE}"
---
schema: olm.channel
package: lightspeed-operator
name: preview
entries:
  - name: lightspeed-operator.v$VERSION
EOF

echo "Building catalog image ${CATALOG_IMAGE}"
podman build --platform="${DEFAULT_PLATFORM}" -f "${CATALOG_DOCKER_FILE}" -t "${CATALOG_IMAGE}" .
echo "Pushing catalog image ${CATALOG_IMAGE}"
podman push "${CATALOG_IMAGE}"
```

Reply:
shall we use the `$CONTAINER_TOOL` env var here?

Reply:
Env var or the var in the script?
But honestly, I (personally) wouldn't switch to `podman` in this PR.
This can be done in a separate PR because we are breaking something (I mentioned [above](https://github.com/openshift/lightspeed-operator/pull/24#discussion_r1510236322))

Reply:
Thanks for the explanation. Let's keep using docker now.

Reply:
I would like to use podman for this script and Keep the Makefile as is . Does that sound okay ?

Reply:
yes, sounds good

Reply:
Updated the script to match versions like 0.0.12 .. multidigits